tier4/Pilot.Auto
# ROS2 Porting: emergency_handler

# emergency_handler

## Summary

Port the emergency_handler package to ROS2. Simple package with not many dependencies so the conversion was straightforward. Key changes:
* Convert headers and type declarations to ROS2
* Convert logging 
* Use of timers
* Conversion of parameter configuration files

## Testing

1. Using the ADE environment in foxy, compile the package
```
colcon build
. install/setup.bash
```
2. Launch the package
```
ros2 launch emergency_handler emergency_handler.launch.xml
```
3. Check to make sure that the configuration file parameters have been loaded correctly
```
ros2 param list
```

## Discussion Points

1. Sorry about the annoying diff view for some of the files, this package had a strange structure to it and I wanted to change, as a result some of the packages have been moved and renamed. If this is an unnecessary change I can revert these easily.
